docs/architecture-decision-records/0022-replace-datasync.md

```diff
@@ -0,0 +1,64 @@
+# Architectural Decision Record (ADR): Switching from AWS DataSync to Direct Lambda-to-Preservica Transfers
+
+## Context
+In our current architecture, the opex creator lambda creates the opex files in our staging AWS S3 bucket. These files are later transferred to Preservica, which uses an S3 bucket in their AWS account.
+
+To facilitate this transfer, we used **AWS DataSync**. This approach was initially chosen to simplify permissions, adhering to the principle of least privilege. Instead of allowing multiple Lambdas to directly access the Preservica bucket, we used a single IAM role for DataSync. However, this approach has revealed significant performance limitations:
+
+- **Slow transfer speed**: DataSync’s transfer rate does not meet our requirements, especially during periods of high file volume.
+- **Concurrency limits**: DataSync has a limit on the number of concurrent tasks, causing bottlenecks when multiple file transfers are queued.
+
+Given these limitations, we evaluated alternative approaches to improve performance while maintaining secure and manageable permissions.
+
+## Decision
+We will replace the use of AWS DataSync with a mechanism where each Lambda function directly transfers files to Preservica’s S3 bucket. This will involve:
+
+1. **Updating Lambda Roles**:
+   Each Lambda will continue to use its own IAM role with scoped-down permissions based on its function.
```

Comment:
You put here "Each Lambda will continue to use **its own** IAM role" but previously, you mentioned there were all using a "single IAM role"; should we remove "continue to"?

Reply:
The single IAM role was for Datasync, not for the lambdas. They each had their own role originally because they didn't need to access Preservica's bucket.